spring-integration-core/src/main/java/org/springframework/integration/support/management/micrometer/MicrometerMetricsCaptor.java

```diff
@@ -0,0 +1,206 @@
+	}
+
+	@Override
+	public SampleFacade start() {
+		return new MicroSample(Timer.start(this.meterRegistry));
+	}
+
+	private class MicroSample implements SampleFacade {
+
+		private final Timer.Sample sample;
+
+		MicroSample(Timer.Sample sample) {
+			this.sample = sample;
+		}
+
+		@Override
+		public void stop(TimerFacade timer) {
+			this.sample.stop(((MicroTimer) timer).timer);
+		}
+
+	}
+
+	private class MicroTimerBuilder implements TimerBuilder {
+
+		private final Timer.Builder builder;
+
+		MicroTimerBuilder(String name) {
+			this.builder = Timer.builder(name);
+		}
+
+		@Override
+		public MicroTimerBuilder tag(String key, String value) {
+			this.builder.tag(key, value);
+			return this;
+		}
+
+		@Override
+		public MicroTimerBuilder description(String desc) {
+			this.builder.description(desc);
+			return this;
+		}
+
+		@Override
+		public MicroTimer build() {
+			return new MicroTimer(this.builder.register(MicrometerMetricsCaptor.this.meterRegistry));
+		}
+
+	}
+
+	private class MicroTimer implements TimerFacade {
+
+		private final Timer timer;
+
+		MicroTimer(Timer timer) {
+			this.timer = timer;
+		}
+
+		@Override
+		public SampleFacade start(MetricsCaptor captor) {
+			return new MicroSample(Timer.start(MicrometerMetricsCaptor.this.meterRegistry));
+		}
+
+		@Override
+		public void record(long time, TimeUnit unit) {
+			this.timer.record(time, unit);
+		}
+
+	}
+
+	private class MicroCounterBuilder implements CounterBuilder {
+
+		private final Counter.Builder builder;
+
+		MicroCounterBuilder(String name) {
+			this.builder = Counter.builder(name);
+		}
+
+		@Override
+		public CounterBuilder tag(String key, String value) {
+			this.builder.tag(key, value);
+			return this;
+		}
+
+		@Override
+		public CounterBuilder description(String desc) {
+			this.builder.description(desc);
+			return this;
+		}
+
+		@Override
+		public CounterFacade build() {
+			return new MicroCounter(this.builder.register(MicrometerMetricsCaptor.this.meterRegistry));
+		}
+
+	}
+
+	private class MicroCounter implements CounterFacade {
+
+		private final Counter counter;
+
+		MicroCounter(Counter counter) {
+			this.counter = counter;
+		}
+
+		@Override
+		public void increment() {
+			this.counter.increment();
+		}
+
+	}
+
+	private class MicroGaugeBuilder implements GaugeBuilder {
+
+		private final Gauge.Builder<Object> builder;
+
+		MicroGaugeBuilder(String name, Object obj, ToDoubleFunction<Object> f) {
+			this.builder = Gauge.builder(name, obj, f);
+		}
+
+		@Override
+		public GaugeBuilder tag(String key, String value) {
+			this.builder.tag(key, value);
+			return this;
+		}
+
+		@Override
+		public GaugeBuilder description(String desc) {
+			this.builder.description(desc);
+			return this;
+		}
+
+		@Override
+		public GaugeFacade build() {
+			this.builder.register(MicrometerMetricsCaptor.this.meterRegistry);
+			return new MicroGauge();
+		}
+
+	}
+
+	private class MicroGauge implements GaugeFacade {
```

Comment:
Can we make all these inner classes as `static`?
I don't see too much overhead with just a ctor arg propagation for the `meterRegistry`.
However it is there at runtime when we create builders instances and have a hard link to outer class.

WDYT?